### ISSUE TITLE
Chunk Context Signal Points

### ISSUE THREAD

**Issue by andrewgalbraith21** [nanoporetech/remora#112]
Hello quick question, 

For the chunk_context argument in dataset prepare, do the number signal points correspond to any estimated number of bases from around a central position? For instance, for the default of +/- 50 context signal points, do you have any estimate of how many bases this might correspond or is it too variable to know

**Comment by marcus1487** [comment]
Our current recommended kit 14 with 5kHz setting runs at 400 bases/sec and collects data at 5kHz. Putting these together gives us 12.5 samples per base. So about 8 bases of signal on average. Estimating the variance in this value is a bit of a trickier problem as the exact assignment of signal will have a large effect. Remora provides a solution for precise signal to sequence assignment, but this implementation is far from perfect (is not used in production modified base detection models). Remora does aim to provide the tools to investigate this though (have a look at the metrics notebook included with the repo.

**Comment by andrewgalbraith21** [comment]
Hello @marcus1487 thank you this is very helpful and will do! As one more follow up, do you have any idea what the estimates for R9 and R10 chemistry are like?

**Comment by marcus1487** [comment]
Speed control on device was only added recently. Before this speed was a bit more variable, so I don't have exact expected levels for older conditions.

**Comment by ZhouShibo0823** [comment]
> Our current recommended kit 14 with 5kHz setting runs at 400 bases/sec and collects data at 5kHz. Putting these together gives us 12.5 samples per base. So about 8 bases of signal on average. Estimating the variance in this value is a bit of a trickier problem as the exact assignment of signal will have a large effect. Remora provides a solution for precise signal to sequence assignment, but this implementation is far from perfect (is not used in production modified base detection models). Remora does aim to provide the tools to investigate this though (have a look at the metrics notebook included with the repo.

Hello, I wonder how to choose the value of the parameter --chunk_context, I have trained the model for my novel modification by setting chunk_context from 50 to 600 and found that the accuarcy on testing dataset increased significantly as that value increased. I wonder if too large chunk_context will influence the traing effect?

**Comment by marcus1487** [comment]
@spoweekkk There are a lot of factors that go into selecting the correct hyper-paramters for training. For the chunk_context parameter the largest concern with increasing this value is that the model will overtrain to the dataset or type of data in your training and will not extrapolate to other native data types. The best test is to have an orthogonal data type to use as your target. Often orthogonal data types do not have solid ground truth and thus cannot be used for training, but you may be able to develop a metric closest to your final biological target to assess whether increasing the chunk_context parameter is leading to overtraining on your training data type. 

I hope this helps, but please post if you have any further questions.

**Comment by ZhouShibo0823** [comment]
> @spoweekkk There are a lot of factors that go into selecting the correct hyper-paramters for training. For the chunk_context parameter the largest concern with increasing this value is that the model will overtrain to the dataset or type of data in your training and will not extrapolate to other native data types. The best test is to have an orthogonal data type to use as your target. Often orthogonal data types do not have solid ground truth and thus cannot be used for training, but you may be able to develop a metric closest to your final biological target to assess whether increasing the chunk_context parameter is leading to overtraining on your training data type.
> 
> I hope this helps, but please post if you have any further questions.

Thanks for your quick reply, and I also want to ask a simple question :
I notice that after "remora dataset prepare",  the chunks have been extracted from the data, and I can find the chunks number in the metadata.jsn file. 
Since I found that no matter how I change  the parameter --chunk_context, the "allocate_size" or the "dataset_end" didn't change anymore. I feel very strange about that. If --chunk_context has changed, doesn't it mean that the length of a chunk changes? If so, why the total number of chunk doesn't change?
